### PR TITLE
#162725252 Document API using swagger

### PR DESCRIPTION
**Pull request by calvinpete**
#### What does this PR do?
This PR is to document the APIs using swagger

#### Description of Task to be completed?
Document the APIs using swagger to create a concise reference manual containing all the information required to work with the APIs

#### How should this be manually tested?
- First things first, fetch this branch by running the command `git fetch origin ft-document-api-162725252` in the terminal.
- Check into this branch by running `git checkout ft-document-api-162725252`
- Activate your virtual environment by running `source venv/bin/activate`
- Install dependencies by running `pip install -r requirements.txt`
- Start the application by running `python manage.py runserver`
- Then view the documentation by navigating to this [link](http://localhost:8000/api/documentation/)

#### What are the relevant pivotal tracker stories?
[#162725252](https://www.pivotaltracker.com/story/show/162725252)

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 )
 
 
-class RegistrationAPIView(APIView):
+class RegistrationAPIView(GenericAPIView):
```
Comment (ja-odur) [inline authors/apps/authentication/views.py:13]: @calvinpete Why the change to using `GenericAPIView`.

File: authors/urls.py [hunk 2]
```diff
@@ -15,8 +15,24 @@
...
+      description="A social platform for the creative at heart.",
+      terms_of_service="https://www.google.com/policies/terms/",
+      contact=openapi.Contact(email="contact@snippets.local"),
+      license=openapi.License(name="BSD License"),
+   ),
+   permission_classes=(AllowAny,),
+)
 
 urlpatterns = [
+    path('swagger/', schema_view.with_ui('swagger', cache_timeout=0), name='schema-swagger-ui'),
```
Comment (ja-odur) [inline authors/urls.py]: @calvinpete I think you should change the path to `documentation/` instead of `swagger/`

File: requirements.txt [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 django-extensions==2.1.4
 djangorestframework==3.9.0
+drf-yasg==1.12.1
+openapi-codec==1.3.2
 psycopg2==2.7.6.1
 psycopg2-binary==2.7.6.1
 PyJWT==1.7.1
 python-decouple==3.1
 pytz==2018.9
+ruamel.yaml==0.15.85
```
Comment (ja-odur) [inline requirements.txt]: @calvinpete did you use `pip freeze` to generate this file. otherwise what does this requirement do.

File: authors/apps/authentication/views.py (hunk 1 above)
Comment (calvinpete) [reply authors/apps/authentication/views.py:13]: The Input parameters do not appear in the swagger documentation when we use `APIView `but the `GenericAPIView` uses the `get_serializer_class()` method to implement the views

**Review by kdaglas — DISMISSED** [review]
This is good though fix the merge conflicts and we see from there

**Comment by armstrongsouljah** [comment]
@calvinpete If the `swagger docs generator` does not have relations it saves in our database, I think it would be okay to remove `python manage.py migrate` in the PR description. Otherwise, the docs work as expected for the available endpoints. Good job on that.

**Review by MuhweziDeo — COMMENTED** [review]
It appears to run fine locally

**Comment by MuhweziDeo** [comment]
> @calvinpete If the `swagger docs generator` does not have relations it saves in our database, I think it would be okay to remove `python manage.py migrate` in the PR description. Otherwise, the docs work as expected for the available endpoints. Good job on that.

right
@armstrongsouljah

File: authors/urls.py (hunk 2 above)
Comment (MuhweziDeo) [reply authors/urls.py]: HE CAN AS WELL AS PUT IT AT '/' since there is nothing there @ja-odur

**Review by armstrongsouljah — DISMISSED** [review]
All looks good to me. Thank you for the good work on the documentations.

**Review by mariamiah — DISMISSED** [review]
Code looks good

File: authors/apps/authentication/urls.py [hunk 4]
```diff
@@ -3,10 +3,25 @@
...
+      description="A social platform for the creative at heart.",
+      terms_of_service="https://www.google.com/policies/terms/",
+      contact=openapi.Contact(email="contact@snippets.local"),
+      license=openapi.License(name="BSD License"),
+   ),
+   permission_classes=(AllowAny,),
+)
 
 urlpatterns = [
     path('user/', UserRetrieveUpdateAPIView.as_view()),
```
Comment (MuhweziDeo) [inline authors/apps/authentication/urls.py:8]: @calvinpete  did you rebase i Made some changes here

Comment (MuhweziDeo) [reply authors/apps/authentication/urls.py:8]: I added `name` for  path

File: authors/urls.py (hunk 2 above)
Comment (armstrongsouljah) [reply authors/urls.py]: Good point @MuhweziDeo so that when our application is loaded, the docs show up on the index route. I hope that makes sense.

**Review by d-rita — APPROVED** [review]

**Review by mariamiah — APPROVED** [review]

**Review by ja-odur — APPROVED** [review]
LGTM

File: authors/apps/authentication/urls.py [hunk 5]
```diff
@@ -4,9 +4,8 @@
     LoginAPIView, RegistrationAPIView, UserRetrieveUpdateAPIView
 )
 
-
 urlpatterns = [
     path('user/', UserRetrieveUpdateAPIView.as_view()),
```
Comment (armstrongsouljah) [inline authors/apps/authentication/urls.py:8]: The code looks okay except for the `UserRetrieveUpdateAPIView` route which does not have a name parameter, thus running `reverse()` on this route will cause this route to break.

**Review by kdaglas — COMMENTED** [review]
its now good to go
